### PR TITLE
AS-542: Use gh actions for unit tests [risk: low]

### PR DESCRIPTION
**Pull request by davidangb**
Use GH Actions for unit tests instead of Travis.

Because the GH Actions -> Coveralls integration for sbt is weak (see for instance https://github.com/lemurheavy/coveralls-public, issue 1388), this PR also:
* switches from `scoverage` to `sbt/sbt-jacoco` to generate coverage reports
* switches from coveralls to codecov as our hosted coverage service

Note that the Travis checks for this PR will fail, because I've removed the .travis.yml; this is expected until we disable Travis for this repo. Once this PR is approved, I'll change which checks are required for PRs.

-----

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've followed [the instructions](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [ ] I've updated the [FISMA documentation](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this **unless it's for a hotfix**. In that case, don't delete it!
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,29 +0,0 @@
...
-after_success: sbt coveralls
-
-notifications:
-  email: false
-
-# https://www.scala-sbt.org/1.x/docs/Travis-CI-with-sbt.html describes how to enable caches for sbt dependencies
-before_cache:
-  - rm -fv $HOME/.ivy2/.sbt.ivy.lock
-  - find $HOME/.ivy2/cache -name "ivydata-*.properties" -print -delete
-  - find $HOME/.sbt        -name "*.lock"               -print -delete
```
Comment (ansingh7115) [inline .travis.yml:23]: it's nice we don't need these steps anymore!

**Comment by ansingh7115** [comment]
Will the "Missing base commit" error be automatically fixed when this is merged - is it just because this is the first time we're using it that we're getting this error?

**Review by ansingh7115 — APPROVED** [review]
Looks like the tests run as they should on GitHub Actions

**Comment by davidangb** [comment]
@ansingh7115 yup, I expect that the "missing base commit" error from codecov will be fixed after this merges and we start to upload ongoing coverage reports to them. If it remains a problem, we'll fix it!

**Pull request merged**